### ISSUE TITLE
Internal compiler error: assertion failed (librbml/lib.rs:462)

### ISSUE THREAD

**Issue by neykov** [rust-lang/rust#25540]
```
error: internal compiler error: unexpected panic
note: the compiler unexpectedly panicked. this is a bug.
note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports
note: run with `RUST_BACKTRACE=1` for a backtrace
thread 'rustc' panicked at 'assertion failed: `(left == right)` (left: `64384`, right: `64389`)', /home/rustbuild/src/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librbml/lib.rs:462

stack backtrace:
   1:     0x2af16b5ef5c9 - sys::backtrace::write::h350af8200010e467Z5r
   2:     0x2af16b5f7519 - panicking::on_panic::h4194b7c3f2c96e6a0Kw
   3:     0x2af16b5b8212 - rt::unwind::begin_unwind_inner::hf945ded7ea7b4f57aqw
   4:     0x2af16b5b8fa7 - rt::unwind::begin_unwind_fmt::h5ee26903db1c6d64gpw
   5:     0x2af1722d6ad4 - reader::doc_as_u64::h909934356f9014c5HNa
   6:     0x2af16d590fc7 - metadata::decoder::translated_def_id::hb10091d16807c034uwj
   7:     0x2af16c33eb7e - build_reduced_graph::GraphBuilder<'a, 'b, 'tcx>::build_reduced_graph_for_item::h7ac77aae88598a46hSa
   8:     0x2af16c33956f - build_reduced_graph::BuildReducedGraphVisitor<'a, 'b, 'tcx>.Visitor<'v>::visit_item::h5827d119cd5f21d5MNb
   9:     0x2af16c38ec6a - resolve_crate::haf19754dce12ebe9iai
  10:     0x2af16b21daa2 - driver::phase_3_run_analysis_passes::heb3d8e9920254d63tGa
  11:     0x2af16b2031ac - driver::compile_input::hd3dd730970eb45c2Qba
  12:     0x2af16b2b91b1 - run_compiler::h4be6ca332f13312675b
  13:     0x2af16b2b6a02 - boxed::F.FnBox<A>::call_box::h7925108573723512372
  14:     0x2af16b2b5fc9 - rt::unwind::try::try_fn::h5378213576640735239
  15:     0x2af16b66fc28 - rust_try_inner
  16:     0x2af16b66fc15 - rust_try
  17:     0x2af16b2b6264 - boxed::F.FnBox<A>::call_box::h12991688188475136491
  18:     0x2af16b5f62b1 - sys::thread::Thread::new::thread_start::h750ed153b2916909hxv
  19:     0x2af171790181 - start_thread
  20:     0x2af16bc0547c - __clone
  21:                0x0 - <unknown>
```

**Comment by geraldstanje** [comment]
how about posting the code to see what causes the error? :)

**Comment by neykov** [comment]
@geraldstanje https://github.com/neykov/armboot/tree/wip-rust-1.0.0-beta

**Comment by colinmccabe** [comment]
I encountered the same error when trying to build https://github.com/0xc0170/frdm-kl25z-rust after upgrading Rust to 1.3.0 nightly (cb7d0621556fabcdcc36d0da279305ad285ec662). It turns out I had neglected to rebuild libcore with the new rustc. After doing that the build completed successfully. Unfortunately I can't figure out what version of Rust was installed before -- it may have been from March 2015. Hope this helps.
